tasks/main.yml

```diff
@@ -6,43 +6,25 @@
     virtualenv: "{{ os_container_clusters_venv or omit }}"
   become: "{{ os_container_clusters_venv == None }}"
 
-- name: Check whether container cluster templates exist
-  shell: >
-    {{ os_container_clusters_venv ~ '/bin/' if os_container_clusters_venv else '' }}openstack
-    {% for auth_name, auth_value in os_container_clusters_auth.items() %}
-    --os-{{ auth_name | replace('_', '-') }}='{{ auth_value }}'
-    {% endfor %}
-    {% if os_container_clusters_cacert is defined %}
-    --os-cacert {{ os_container_clusters_cacert }}
-    {% endif %}
-    --os-interface=public
-    coe cluster template show {{ item.name }} >/dev/null
+- name: Ensure container cluster templates exist
+  vars:
+    ansible_python_interpreter: "{{ os_container_clusters_venv ~ '/bin/python' if os_container_clusters_venv != None else old_ansible_python_interpreter }}"
```

Comment:
@markgoddard like this?

Reply:
Yes, but you need to define old_ansible_python_interpreter. See how it's done in other roles.